@xen-orchestra/web/src/types/xo/pif.type.ts

```diff
@@ -0,0 +1,21 @@
+import type { XoHost } from '@/types/xo/host.type'
+import type { XoNetwork } from '@/types/xo/network.type'
+
+export type XoPif = {
+  $host: XoHost['id']
+  $network: XoNetwork['id']
+  attached: boolean
+  carrier: boolean
+  device: string
+  dns: string
+  gateway: string
+  id: string
```

Comment:
Remember to _brand_ the objects' id:
```suggestion
  id: Branded<'pif'>
```